variables.tf

```diff
@@ -243,3 +243,17 @@ variable "license_type" {
     error_message = "License type must be \"None\", \"Windows_Client\", \"Windows_Server\", \"RHEL_BYOS\" or \"SLES_BYOS\"."
   }
 }
+
+variable "os_disk_encryption_set_id" {
+  description = "(Optional) The ID of the Disk Encryption Set which should be used to Encrypt this OS Disk."
+  type        = string
+  default     = null
+  nullable    = true
+}
+
+variable "data_disk_encryption_set_id" {
+  description = "(Optional) The ID of the Disk Encryption Set which should be used to Encrypt data Disks."
+  type        = string
+  default     = null
+  nullable    = true
+}
```

Comment:
I think this should be configurable on a per-disk-basis, i.e. it this standalone variable should be removed and an property `disk_encryption_set_it` of type `optional(string)` should be added to this existing variable instead:

https://github.com/equinor/terraform-azurerm-vm/blob/7ba88873c71d2223e32201997792a196be13c9c3/variables.tf#L190-L202

Reply:
Good input. Implemented.